node/engine/messageservice/p2p-message-service/service.go

```diff
@@ -143,19 +142,21 @@ func (ms *P2PMessageService) setupDht(bootPeers []string) {
 	addrs, err := peer.AddrInfoToP2pAddrs(&peerInfo)
 	ms.checkError(err)
 	ms.MultiAddr = addrs[0].String()
-	ms.logger.Debug().Msgf("libp2p node address: %s", ms.MultiAddr)
+	ms.logger.Debug().Msgf("libp2p node multiaddrs: %v", addrs)
+
+	// Setup notifications so we exchange nitro signing addresses when connected
+	n := &NetworkNotifiee{ms: ms}
+	ms.p2pHost.Network().Notify(n)
 
-	// Add bootstrap peers
-	err = ms.dht.Bootstrap(ctx)
-	ms.checkError(err)
 	ms.addBootPeers(bootPeers)
 
 	expectedPeers := len(bootPeers)
 	if expectedPeers > 0 {
+		ms.logger.Debug().Msgf("waiting for %d bootpeer connections", expectedPeers)
 		ticker := time.NewTicker(BOOTSTRAP_SLEEP_DURATION)
 		for range ticker.C {
-			peers := ms.p2pHost.Peerstore().Peers()
-			actualPeers := len(peers) - 1 // subtract one to ignore self
+			peers := ms.p2pHost.Network().Peers()
```

Comment:
This line checks the actual number of peers this node is currently connected. `ms.p2pHost.Peerstore().Peers()` checks how many entries are in the local `Peerstore` which could be leftover data from a previous node session if the node is using a persistent store. Also an entry in the `Peerstore` does not mean there is an actual connection between the nodes.